### PR TITLE
build(deps): update github.com/prometheus/prometheus to v0.42.0

### LINKED ISSUE #2568: Handle github.com/prometheus/prometheus v0.42.0 breaking changes
prometheus/prometheus#11717 has reworked its Go API for managing labels, creating breaking changes in the v0.42.0 release. Parca Server and Agent need to be update to use this better abstracted API.

### PR DESCRIPTION
**Pull request by maxbrunet**
Closes #2568

### CODE REVIEW

File: pkg/parca/parca.go [hunk 1]
```diff
@@ -584,10 +584,7 @@ func runScraper(
 		return err
 	}
 
-	externalLabels := labels.Labels{}
-	for name, value := range flags.ExternalLabel {
-		externalLabels = append(externalLabels, labels.Label{Name: name, Value: value})
-	}
+	externalLabels := labels.FromMap(flags.ExternalLabel)
```
Comment (kakkoyun) [inline pkg/parca/parca.go:587]: For reference: https://github.com/prometheus/prometheus/blob/2d588725c12e46e4e71da3899ec05bb933151021/model/labels/labels.go#L367

File: pkg/scrape/scrape.go [hunk 2]
```diff
@@ -216,9 +216,9 @@ func (sp *scrapePool) Sync(tgs []*targetgroup.Group) {
 		}
 
 		for _, t := range targets {
-			if t.Labels().Len() > 0 {
+			if !t.Labels().IsEmpty() {
```
Comment (kakkoyun) [inline pkg/scrape/scrape.go:219]: For reference: https://github.com/prometheus/prometheus/blob/2d588725c12e46e4e71da3899ec05bb933151021/model/labels/labels.go#L421

File: pkg/scrape/scrape.go [hunk 3]
```diff
@@ -468,12 +468,12 @@ mainLoop:
 
 			tl := sl.target.Labels()
 			tl = append(tl, labels.Label{Name: "__name__", Value: profileType})
-			for _, l := range sl.externalLabels {
+			sl.externalLabels.Range(func(l labels.Label) {
```
Comment (kakkoyun) [inline pkg/scrape/scrape.go:471]: For reference:
https://github.com/prometheus/prometheus/blob/2d588725c12e46e4e71da3899ec05bb933151021/model/labels/labels.go#L426

File: pkg/scrape/target.go [hunk 4]
```diff
@@ -108,22 +108,20 @@ func (t *Target) Params() url.Values {
 
 // Labels returns a copy of the set of all public labels of the target.
 func (t *Target) Labels() labels.Labels {
-	lset := make(labels.Labels, 0, len(t.labels))
-	for _, l := range t.labels {
+	b := labels.NewScratchBuilder(t.labels.Len())
```
Comment (kakkoyun) [inline pkg/scrape/target.go:111]: For reference:
https://github.com/prometheus/prometheus/blob/2d588725c12e46e4e71da3899ec05bb933151021/model/labels/labels.go#L576

File: pkg/scrape/target.go [hunk 5]
```diff
@@ -206,19 +204,16 @@ func (t *Target) Health() TargetHealth {
...
-		}
-	}
 
-	if c.PprofConfig != nil {
+	if len(c.PprofConfig) > 0 {
+		b := labels.NewScratchBuilder(lset.Len() + 2)
 		for profilingType, profilingConfig := range c.PprofConfig {
-			add(profilingType, *profilingConfig)
+			if *profilingConfig.Enabled {
+				b.Assign(lset)
```
Comment (kakkoyun) [inline pkg/scrape/target.go:212]: Ref: https://github.com/prometheus/prometheus/blob/2d588725c12e46e4e71da3899ec05bb933151021/model/labels/labels.go#L596

**Review by kakkoyun — APPROVED** [review]
LGTM ❤️ 💚 💙 💛 💜

**Review by metalmatze — APPROVED** [review]
Amazing! Thanks for taking the time! :tada: 
Fewer changes for this than I expected. :upside_down_face:

**Pull request merged**